lib/optimizely/helpers/validator.rb

```diff
@@ -108,10 +108,12 @@ def inputs_valid?(variables, logger = NoOpLogger.new, level = Logger::ERROR)
         return false unless variables.respond_to?(:each) && !variables.empty?
         is_valid = true
         variables.each do |key, value|
-          next if value.is_a?(String) && !value.empty?
-          is_valid = false
-          if logger_valid?(logger) && level
-            logger.log(level, "#{Optimizely::Helpers::Constants::INPUT_VARIABLES[key.to_s.upcase]} is invalid")
+          is_valid = false unless value.is_a?(String)
+          unless key.to_s.upcase == Helpers::Constants::INPUT_VARIABLES.key('User ID')
```

Comment:
This is not the right way. The method should only do it's own task. It should not know if the incoming thing is a user ID or not. You can keep this method as it is. And for user ids, check string right at the place